### PR TITLE
[Folder] Add general UID validation

### PR DESCRIPTION
**Pull request by leonorfmartins**
To be compliant withe the legacy implementation, we need to make sure that the legacy UID aka name in storage doesn't have the value "general".

**What is this feature?**

Part of https://github.com/grafana/search-and-storage-team/issues/110

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Comment by leonorfmartins** [comment]
discussion about it: https://raintank-corp.slack.com/archives/C04RVCAG9B5/p1730202076466669

File: pkg/registry/apis/folders/register_test.go [hunk 1]
```diff
@@ -114,3 +117,51 @@ func TestFolderAPIBuilder_getAuthorizerFunc(t *testing.T) {
...
 	}
 }
+
+func TestFolderAPIBuilder_Validate(t *testing.T) {
+	tests := []struct {
+		name  string
+		input *unstructured.Unstructured
+		err   error
+	}{
+		{
```
Comment (suntala) [inline pkg/registry/apis/folders/register_test.go:127]: Do we want to add a test case for reaching max depth as well?

**Review by suntala — COMMENTED** [review]
Really nice! Just a couple of comments

File: pkg/registry/apis/folders/register.go [hunk 2]
```diff
@@ -220,3 +218,38 @@ func authorizerFunc(ctx context.Context, attr authorizer.Attributes) (*authorize
...
+
+func (b *FolderAPIBuilder) Validate(ctx context.Context, a admission.Attributes, _ admission.ObjectInterfaces) error {
+	// name cannot be called "general"
+	id := a.GetName()
+	if id == invalidName {
+		return dashboards.ErrFolderInvalidUID
+	}
+
+	//max depth folder is 4
+	var maxFolderDepth = 4
```
Comment (suntala) [inline pkg/registry/apis/folders/register.go:230]: nit: Isn't this defined as a variable somewhere else? We could use it here as well to ensure we remain consistent.

Comment (suntala) [reply pkg/registry/apis/folders/register.go:230]: Seeing Bruno's comment above, same thing could be said for the constant for the General folder name.

File: pkg/registry/apis/folders/register_test.go (hunk 1 above)
Comment (leonorfmartins) [reply pkg/registry/apis/folders/register_test.go:127]: yes, but right now we can't add the test because we aren't validating it yet :)

**Review by inf0rmer — APPROVED** [review]

**Pull request merged**